tests/test_misc.py

```diff
@@ -176,3 +186,28 @@ def test__deep_update__overwrite_strings() -> None:
         },
         "deny": {},
     }
+
+
+def test_deprecated_logs_warning_for_function(caplog: LogCaptureFixture) -> None:
+    caplog.set_level(logging.WARNING)
+    expected_warning = (
+        "Call to deprecated function smth. dont use me. Function will be removed in 9.9.9."
+    )
+
+    with pytest.warns(DeprecationWarning, match=expected_warning):
+
+        @deprecated("dont use me", "9.9.9")
```

Comment:
According to the first comment: the warning should be shown to the user only when the function is used, not when declared.